File: Hotels/Models/Invoice.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Hotels.Models
{
    public class Invoice
    {
        public int Id { get; set; }

        [Required]
        public double TotalAmount { get; set; }

        public bool IsPaid { get; set; } = false;

        public virtual Reservation Reservation { get; set; }

        public virtual ICollection<Item> Items { get; set; }
    }
}

File: Hotels/Scripts/Invoice.js
function Invoice(object) {
    if (object) {
        this.id = object.Id;
        this.reservationId = object.reservationId;
        this.isPaid = object.isPaid;
        this.totalAmount = object.totalAmount;
        this.items = object.items;
    } else {
        this.id = null;
        this.reservationId = null;
        this.isPaid = null;
        this.totalAmount = null;
        this.items = null;
    }
}

Invoice.prototype.toString = function () {
    return this.id + ": " + this.isPaid;
}